pkg/odo/cli/component/create.go

```diff
@@ -71,11 +71,11 @@ const CreateRecommendedCommandName = "create"
 // since the application will always be in the same directory as `.odo`, we will always set this as: ./
 const LocalDirectoryDefaultLocation = "./"
 
-// DevfilePath is the default path of devfile.yaml
-const DevfilePath = "./devfile.yaml"
+// DevfilePath is the path of devfile.yaml, the default path is "./devfile.yaml"
+var DevfilePath = filepath.Join(LocalDirectoryDefaultLocation, "devfile.yaml")
```

Comment:
IIRC some functions still need that, but this could become local variable in my `--devfile` support PR since those functions can always use `./devfile.yaml` as devfile path once `--devfile` support is implemented, I can address that later.